### PR TITLE
Added searching and browsing use cases

### PR DESCRIPTION
**Pull request by westonh**
Added a few searching and browsing use cases to use-cases.md. Primarily, I want to make sure that we build an experience that can work in both CLI and UI environments.

### CODE REVIEW

File: docs/v1/use-cases.md [hunk 1]
```diff
@@ -32,6 +44,64 @@ Kubernetes cluster to publish their services using a service broker.  This
...
+    * image URLs - a list of images that could be displayed in a UI
+    * TOS link
+    * a list of plans
+        * plan name
+        * plan description
+        * plan cost
+    * construction parameters
+        * name
+        * description
+        * default value
```
Comment (arschles) [inline docs/v1/use-cases.md]: Based on the above item 3 under "Searching and Browsing Services", it seems like there should also be an "optional" parameter

File: docs/v1/use-cases.md [hunk 2]
```diff
@@ -16,6 +16,18 @@ implement.
...
+2. Searching and Browsing Services:
+    1.  As a consumer, I'm able to search my catalog for services by attributes
+        such as category.
+    2.  As a consumer, I'm able to see metadata about a service prior to 
+        creation which allows me to see if this service fits my need.
+    3.  As a consumer, I'm able to see all the required and optional parameters
+        I'll need to pass to the service in order to create it.
+    4.  As a consumer, when listing services I see a union of the catalogs
+        from all brokers I have registered. However, if I want to restrict the
+        list to a specific broker I can pass that in as a flag.
```
Comment (duglin) [inline docs/v1/use-cases.md]: I'm ok with call this out as a separate usecase, but it feels like a subset of the first one.

File: docs/v1/use-cases.md [hunk 3]
```diff
@@ -32,6 +44,64 @@ Kubernetes cluster to publish their services using a service broker.  This
...
+We should consider what metadata needs to be exposed for a strong CLI and UI
+experience. Here are some suggestions for metadata fields:
+
+    * name
+    * short description
+    * long desciption
+    * documentation/support urls
+    * icon URL
+    * image URLs - a list of images that could be displayed in a UI
+    * TOS link
```
Comment (duglin) [inline docs/v1/use-cases.md]: TOS?

**Comment by duglin** [comment]
CLA needed

File: docs/v1/use-cases.md (hunk 3 above)
Comment (pmorie) [inline docs/v1/use-cases.md]: I read that as 'terms of service'

File: docs/v1/use-cases.md (hunk 2 above)
Comment (pmorie) [inline docs/v1/use-cases.md]: Agree

**Comment by pmorie** [comment]
@kubernetes-incubator/maintainers-service-catalog PTAL, let's get this merged unless there are remaining reservations.

File: docs/v1/use-cases.md (hunk 3 above)
Comment (duglin) [inline docs/v1/use-cases.md]: can we expand "TOS" ?

**Comment by duglin** [comment]
One minor change request, and rebase needed.

**Comment by pmorie** [comment]
@westonh would you rebase this, and we'll get it merged?

**Comment by arschles** [comment]
@westonh what's the status of this PR?

**Comment by pmorie** [comment]
@westonh and i discussed someone else cherry-picking this commit in and
rebasing it.  Go for it if you have time -- I probably won't get to it
until tomorrow.


On Tue, Dec 6, 2016 at 12:53 PM, Aaron Schlesinger <notifications@github.com

**Comment by pmorie** [comment]
I'm closing this and will open a new PR with content rebased and nits addressed.

**Pull request closed without merging**